core/blockchain.go

```diff
@@ -223,6 +223,22 @@ func (bc *BlockChain) ValidateNewBlock(block *types.Block, address common.Addres
 	return nil
 }
 
+// IsEpochBlock returns whether this block is the first block of an epoch.
+func IsEpochBlock(block *types.Block) bool {
+	if block.NumberU64()%BlocksPerEpoch == 0 {
+		return true
+	}
+	return false
+}
+
+// IsEpochLastBlock returns whether this block is the last block of an epoch.
```

Comment:
maybe move this to core/resharding.go ?

Reply:
I kept both here and switched yours to mine.